server/services/authService.js

```diff
@@ -0,0 +1,13 @@
+import { authDatabase } from '../databases';
+
+const handleLogin = async (username, password) => {
+    const { user, sessionId } = await authDatabase.performLogin(username, password);
+
+    // Here we should be saving sessions into a database, using the sessionID returned form performLogin
```

Comment:
This is a big comment, but the `session` logic that existed before needs to change. It's fine for working locally, but it won't work in a hosted environment. There's two reasons here, and both stem from holding **state in memory**. By setting session as an array in the server, we're keeping it in RAM for later use. This is fine, until we move into hosting. 

Most hosts (heroku, digital ocean, kubernetes, etc) all use _containers_ to run applications, and containers are inherantly ephemeral (they come into and out of existance as needed). For example, on heroku if you don't use your application for a few hours the dyno (container) will "go to sleep". When you request again, it will spin up a new instance. This processes clears everything in RAM, and effectively wipes out your all of your session.

What I recommend here is to save session in the database. Mongo is fine for this now, but there are specialized databases for things like this (Redis is a good example). If you're interested in learning about that, we can set it up later!